File: app/src/main/java/com/example/emerus/Details.java
package com.example.emerus;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.bumptech.glide.Glide;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.FirebaseFirestore;

public class Details extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature( Window.FEATURE_NO_TITLE );
        setContentView(R.layout.activity_details);

        // Getting var from MainActivity
        Bundle extras = getIntent().getExtras();
        String profile_name = extras.getString("profile_name");
        TextView profile = findViewById(R.id.profile);
        profile.setText(profile_name);

        // Getting data from Firebase
        FirebaseFirestore db = FirebaseFirestore.getInstance();
        DocumentReference docRef = db.collection("Profili").document(profile_name);
        docRef.get().addOnSuccessListener(documentSnapshot -> {
            String link = (String) documentSnapshot.get("link");
            String vlasnik =(String) documentSnapshot.get("vlasnik");

            TextView vlasniktxt;
            vlasniktxt = findViewById(R.id.vlasniktxt);
            vlasniktxt.setText(vlasnik);

            ImageView crtez;
            crtez = findViewById(R.id.image12);
            Glide.with(Details.this).load(link).into(crtez);

             Button button = findViewById(R.id.button);
            if(link != ""){
            button.setOnClickListener(new View.OnClickListener() {
                public void onClick(View v) {
                    Intent intent = new Intent();
                    intent.setAction(Intent.ACTION_VIEW);
                    intent.addCategory(Intent.CATEGORY_BROWSABLE);
                    intent.setData(Uri.parse(link));
                    startActivity(intent);
                 }}
            );}
            if (link == null){
                button.setOnClickListener(new View.OnClickListener() {
                    public void onClick(View v) {
                        Toast.makeText(Details.this, "Nažalost, nema ništa za preuzimanje. Pokušajte s drugim profilom.", Toast.LENGTH_LONG).show();
                    }});
            };

        });
    }}

File: app/src/main/java/com/example/emerus/Details.kt
package com.example.emerus

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Window
import android.widget.ImageView
import com.example.emerus.R
import android.widget.TextView
import com.google.firebase.firestore.FirebaseFirestore
import com.google.android.gms.tasks.OnSuccessListener
import com.google.firebase.firestore.DocumentSnapshot
import com.bumptech.glide.Glide

class Details : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        val decorView = window.decorView
        setContentView(R.layout.activity_details)
        val extras = intent.extras
        val value = extras!!.getString("profile_name")
        val profile = findViewById<TextView>(R.id.profile)
        profile.text = value

        // Getting data from Firebase
        val db = FirebaseFirestore.getInstance()
        val docRef = db.collection("Profili").document(value!!)
        docRef.get().addOnSuccessListener { documentSnapshot ->
            val link = documentSnapshot["link"] as String?
            val vlasnik = documentSnapshot["vlasnik"] as String?

            val vlasniktxt: TextView
            vlasniktxt = findViewById(R.id.vlasniktxt)
            vlasniktxt.text = vlasnik

            val crtez: ImageView
            crtez = findViewById(R.id.image12)
            Glide.with(this@Details).load(link).into(crtez)
        }
    }
}